src/js/cc_script.js

```diff
@@ -5,139 +5,139 @@
+            dynamicWidth: $('.dynamic').width(), //initializing the dynamic wrapper width
+            sliderLength: $('.slider div').length // count of images in the slider
+        };
+
+        var settings = $.extend(defaults, $.fn.ccCarousel.defaults, options);
+
+        function initConfig() {
+            settings.customWidth = settings.sliderLength * (settings.imageWidth < 50 ? 50 : settings.imageWidth); //the length of the slider div depending on the count of images
+            settings.sliderWidth = settings.customWidth; //slider width in pixels
+            settings.addition = settings.customWidth - settings.wrapperWidth; //the differennce between monitor reslutions
+            settings.clickStep = settings.customWidth / parseInt(settings.sliderLength);
+            settings.slider.width(settings.sliderWidth); //initializing the slider width
+        }
+
+        /*==================================main functions for click====================*/
+
+        function stop() {
+            settings.slider.stop(true);
+        }
+
+        function moveToLeft() {
+            stop();
+            var left = settings.slider.css('left');
+            var leftMargin = parseInt(left);
+
+            if (leftMargin <= -settings.clickStep) {
+                settings.slider.animate({left: leftMargin + settings.clickStep}, settings.clickTimer);
+            }
+            else {
+                settings.slider.animate({left: 0}, settings.clickTimer);
+            }
+        }
+
+        function moveToRight() {
+            stop();
+            var right = settings.slider.css('left');
+            var rightMargin = parseInt(right);
+
+            var customStep = settings.addition + rightMargin;
+
+            if (customStep >= settings.clickStep) {
+                settings.slider.animate({left: rightMargin - settings.clickStep}, settings.clickTimer);
+            }
+            else {
+                settings.slider.animate({left: -settings.addition}, settings.clickTimer);
+            }
+        }
+
+        function leftMouseover() {
+            stop();
+            var leftHover = settings.slider.css('left');
+            var leftMarginHover = parseInt(leftHover);
+
+            if (leftMarginHover !== 0) {
+                settings.slider.animate({left: 0}, settings.hoverTimer);
+            }
+        }
+
+        function leftMouseup() {
+            var leftHover = settings.slider.css('left');
+            var leftMarginHover = parseInt(leftHover);
+
+            if (leftMarginHover !== 0) {
+                settings.slider.animate({left: 0}, settings.hoverTimer);
+            }
+        }
+
+        /*================================Functions that control slider when hovered==============================*/
+
+        function rightMouseover() {
+            settings.slider.stop();
+            settings.slider.animate({left: -settings.addition}, settings.hoverTimer);
```

Comment:
@rhanesoghlyan duplication of statements  is not a good practice please move to some function called `animate`